contractcourt/htlc_outgoing_contest_resolver.go

```diff
@@ -78,24 +85,32 @@ func (h *htlcOutgoingContestResolver) Resolve() (ContractResolver, error) {
 
 	// We'll quickly check to see if the output has already been spent.
 	select {
-	// If the output has already been spent, then we can stop early and
-	// sweep the pre-image from the output.
+	// If the output has already been spent, then we can stop early.
 	case commitSpend, ok := <-spendNtfn.Spend:
 		if !ok {
 			return nil, errResolverShuttingDown
 		}
 
-		// TODO(roasbeef): Checkpoint?
-		return h.claimCleanUp(commitSpend)
+		// If the output was successfully spent and the pre-image was
+		// revealed we'll sweep it from the output.
+		if isSuccessSpend(commitSpend,
+			h.htlcResolution.SignedTimeoutTx != nil) {
+			// TODO(roasbeef): Checkpoint?
+			return h.claimCleanUp(commitSpend)
+		}
+
+		// The output was spent without revealing a pre-image. This
+		// happens when a retribution transaction was sent.
+		return h.emergencyCleanUp()
```

Comment:
I am wondering what happens if you'd delete this block completely? It is also present in the select below. Or do things go wrong after a restart where we trigger on the initial block while the output was already spent? Probably depends on handling the follow-up resolver.